hydra-plutus/src/Hydra/Contract/Head.hs

```diff
@@ -350,6 +352,24 @@ checkContest ctx@ScriptContext{scriptContextTxInfo} contestationDeadline parties
       UpperBound (Finite time) _ ->
         traceIfFalse "upper bound beyond contestation deadline" $ time <= contestationDeadline
       _ -> traceError "contest: no upper bound defined"
+
+  mustNotChangeParameters =
+    traceIfFalse "changed parameters" $
+      parties' == parties
+        && contestationDeadline' == contestationDeadline
+        && headId' == headId
+
+  (contestSnapshotNumber, contestUtxoHash, parties', contestationDeadline', headId') =
+    case fromBuiltinData @DatumType $ getDatum (continuingDatum ctx) of
```

Comment:
We should probably also mark this `fromBuiltinData` as `XXX` as it is a substantial source of script size.